lldb/source/Plugins/TypeSystem/Swift/TypeSystemSwiftTypeRef.cpp

```diff
@@ -2128,8 +2128,69 @@ TypeSystemSwiftTypeRef::GetByteStride(opaque_compiler_type_t type,
 
 lldb::Encoding TypeSystemSwiftTypeRef::GetEncoding(opaque_compiler_type_t type,
                                                    uint64_t &count) {
-  return m_swift_ast_context->GetEncoding(ReconstructType(type), count);
+  if (!type)
+    return lldb::eEncodingInvalid;
+
+  auto impl = [&]() -> lldb::Encoding {
+    count = 1;
+
+    using namespace swift::Demangle;
+    Demangler dem;
+    auto *node = DemangleCanonicalType(dem, type);
+    auto kind = node->getKind();
+
+    if (kind == Node::Kind::BuiltinTypeName) {
+      assert(node->hasText());
+      if (node->getText().startswith(swift::BUILTIN_TYPE_NAME_INT) ||
+          node->getText() == swift::BUILTIN_TYPE_NAME_WORD)
+        return lldb::eEncodingSint;
+      if (node->getText() == swift::BUILTIN_TYPE_NAME_FLOAT)
+        return lldb::eEncodingIEEE754;
+      if (node->getText() == swift::BUILTIN_TYPE_NAME_RAWPOINTER ||
+          node->getText() == swift::BUILTIN_TYPE_NAME_NATIVEOBJECT ||
+          node->getText() == swift::BUILTIN_TYPE_NAME_UNSAFEVALUEBUFFER ||
+          node->getText() == swift::BUILTIN_TYPE_NAME_BRIDGEOBJECT)
+        return lldb::eEncodingUint;
```

Comment:
Any idea what `swift::BUILTIN_TYPE_NAME_VEC` might be?

Reply:
If we don't know, then leaving it as the default is fine, or we could assert on it (and leave it on the default path) so we can more easily find a testcase.

Reply:
`SwiftASTContext` treats `swift::TypeKind::BuiltinVector` by returning `eEncodingInvalid`, so this implementation matches that, but implicitly (by not mentioning vector). Should it explicitly mention `BUILTIN_TYPE_NAME_VEC`?

Reply:
I think that would be a good idea, then it's obvious that we didn't forget about it, but that it is deliberate.

Reply:
Let me know what you think of 5cbe02a